llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -14195,6 +14195,66 @@ static SDValue widenAbs(SDNode *Extend, SelectionDAG &DAG) {
   return DAG.getZExtOrTrunc(NewAbs, SDLoc(Extend), VT);
 }
 
+// Try to widen the build vector and bitcast it to the type of zext.
+// This is a special case for the 128-bit vector types. Intention is to remove
+// the zext and replace it with a bitcast the wider type. While lowering
+// the bitcast is removed and extra commutation due to zext is avoided.
```

Comment:
DAG format comment of the matched pattern

Reply:
@arsenm Add such comment like  zext(build_vec) -> bitcast(build_vec). which visualize the transformation, right?